src/checkers/inference/BaseInferenceResult.java

```diff
@@ -0,0 +1,39 @@
+package checkers.inference;
+
+import checkers.inference.model.Constraint;
+
+import javax.lang.model.element.AnnotationMirror;
+import java.util.Collection;
+import java.util.Map;
+
+public class BaseInferenceResult implements InferenceResult {
+
+    protected final Map<Integer, AnnotationMirror> inferredResults;
```

Comment:
I personally like the previous name `varIdToAnnotation`, as it self-explained this map --- a map from varId to it's inferred annotation solution.